### ISSUE TITLE
Maintenance Window settings - use Tags to identify Monitors

### ISSUE THREAD

**Issue by gardium90** [louislam/uptime-kuma#3121]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

In creating various maintenance window configurations in the UI, it would seem a good option to be able to use tags instead of "specific monitor" selection (or perhaps an option of both?). In my cases, I'm dealing with multiple platforms/services and different environments per platform/service. So I'm also proposing that a combination of tags may be used (not just one). E.g. if one tag is chosen, say a platform/service tag, then all environments of this platform/service would end up in maintenance mode. But if two tags are used, there must be a match to specific platform/service and environment tag (or any other category of tags used). There could be more scenarios where even more tags should be used, so this should hopefully be possible to allow the user to select 'infinite' number of tags that exist on the monitors.

Every time I add a new monitor anywhere today, I'd have to go in and manually add this monitor to any necessary maintenance window configuration to apply. But if the new monitor had the appropriate set of tags with this new feature, it would then automatically be picked up by the maintenance window config if tags were used.

### ✔️ Solution

Say I have 3 services, and 3 environments per service. Let's call them 'A', 'B', 'C' for the services, and 'Dev', 'UAT', and 'PROD' for the various environments. Depending on the type of maintenance/issues, it may affect any combination of these services and environments. By grouping together the necessary tags, e.g. 'A' for all of A's environments, or 'PROD' for all PROD systems, or a combination, say 'A' + 'UAT', then the maintenance window knows which monitors to deactivate during the maintenance window, without having to select specific monitors

### ❓ Alternatives

I haven't found anything that would potentially manage this type of a solution for the maintenance window configs

**Comment by CommanderStorm** [comment]
Is this a duplicate of #2457?
If no, please edit your feature request to make it more distinct and/or how the change you want builds upon the linked issue.
If yes, please close this issue (duplicate issues just makes managing issues harder)

**Comment by gardium90** [comment]
Apologies. I didn't find this feature request while searching keywords in the Issue Tracker.

However, after reading that feature request, there is a major differences in my opinion that I already address in my request, plus in terms of functionality and 'description' of what I want to achieve, my feature request is more specific.

The major difference I see, is that they request "[...] all monitors with any of the tags selected, will be added to the maintenance windows". I specify in this request I'd like an 'all-inclusive match', due to multi instance tagging AND service tagging.

I hope this clarifies, but if you disagree to my point then please let me know and I'll close the request.

Thank you, and have a nice day.

**Comment by CommanderStorm** [comment]
The reason I like the other issue more is, that it is more concise, readable, older and more liked ([we use 👍🏻 to prioritise work](https://github.com/TUM-Dev/Campus-Android/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)).

I don't understand what you mean by “all-inclusive match”. What do you mean?
> The major difference I see, is that they request "[...] all monitors with any of the tags selected, will be added to the maintenance windows". I specify in this request I'd like an 'all-inclusive match', due to multi instance tagging AND service tagging.

Could this not be integrated into the issue @peschmae created?

**Comment by peschmae** [comment]
I think adding a toggle to switch between - `and` / `or` matching for the selected tags shouldn't be too much of an issue.

I can easily add this to my proposal

**Comment by CommanderStorm** [comment]
> I'd like an 'all-inclusive match', due to multi instance tagging AND service tagging

What do you mean by this statement?
What is an instance, and what is a service?

Background: Afaik, we use neither of these terms and only use tags to tag monitors

**Comment by gardium90** [comment]
@CommanderStorm Hello, sorry I didn't have time to respond to your comment from 2 days ago. I thought this request was put to inactive as part of being consolidated with @peschmae 's proposal.

I believe upon reviewing your changes to the request, you seem to have understood what I meant. Service and instance seem to be reflected correctly, and my wish was the appropriate "and-ing of tags" as you have clarified now. I still see a reason to have "or-ing" as well, so however you chose to consolidate the requests I'll just be happy if this feature makes in into a upstream release.

Thank you again for the consideration, have a merry festive season!

**Comment by sevmonster** [comment]
Yes, as someone that uses tags for tons of services, sometimes they intersect. For example, if I will be performing maintenance on Docker on a specific host, then I would want monitors with the `host1` *AND* `docker` tags to be included. But if I am taking down two different hosts for maintenance, I would want monitors `host1` *OR* `host2` to be included.

The fact that there is no capability of using tags for maintenance makes the feature excessively cumbersome for users with a lot of monitors (like me).

**Comment by bmbvenom** [comment]
Hello, it's a year and half later and I was just wondering if there has been any progress on this. If not in v1 maybe in v2?

Thank you for all your work on this great project!

**Comment by CommanderStorm** [comment]
Nobody has done work on this to our knowlege.
It is also not on my cutting block of issues I want to implement => feel free to implement it.

If the UX is close to the rest and does not incurr masssive overhead for the workflow this would be something I would merge.
